src/frontends/pytorch/src/transforms/aten_getitem_replacer.cpp

```diff
@@ -95,6 +95,17 @@ AtenGetItemReplacer::AtenGetItemReplacer() {
             }
             return true;
         }
+        if (auto shape_of = std::dynamic_pointer_cast<opset10::ShapeOf>(input_node)) {
```

Comment:
I think we can support the case when `__getitem__` is located in body and input is located outside. But there is a problem when `ListConstruct` is located outside. Basically we can translate `__getitem__` to `Gather` if input is `Parameter` and `ListConstruct` (with non zero number of inputs) to `Concat` with axis 0. That should cover majority of cases, but we can still have some issues with that approach.

Reply:
moved to translators